### PR TITLE
Core/Handlers: Lock player queues for Arena Team matches

### PR DESCRIPTION
**Pull request by Gooyeth**
**Changes proposed:**

Prevent the characters be removed of arena teams durings queues

**Target branch(es):** 3.3.5

**Issues addressed:** #21487

**Tests performed:** Build.

Please test in game

### CODE REVIEW

File: src/server/game/Handlers/ArenaTeamHandler.cpp [hunk 1]
```diff
@@ -232,6 +233,14 @@ void WorldSession::HandleArenaTeamLeaveOpcode(WorldPacket& recvData)
         SendArenaTeamCommandResult(ERR_ARENA_TEAM_QUIT_S, "", "", ERR_ARENA_TEAM_LEADER_LEAVE_S);
         return;
     }
+
```
Comment (Jildor) [inline src/server/game/Handlers/ArenaTeamHandler.cpp]: Whitespaces

File: src/server/game/Handlers/ArenaTeamHandler.cpp [hunk 2]
```diff
@@ -263,6 +272,11 @@ void WorldSession::HandleArenaTeamDisbandOpcode(WorldPacket& recvData)
         if (arenaTeam->GetCaptain() != _player->GetGUID())
             return;
 
+        // Teams cannot be disbanded during queues
+        BattlegroundQueueTypeId bgQueue = BattlegroundMgr::BGQueueTypeId(BATTLEGROUND_AA, arenaTeam->GetType());
+        if (bgQueue && _player->InBattlegroundQueueForBattlegroundQueueType(bgQueue))
+            return;
+
```
Comment (Jildor) [inline src/server/game/Handlers/ArenaTeamHandler.cpp]: Whitespaces ;)

File: src/server/game/Handlers/ArenaTeamHandler.cpp [hunk 3]
```diff
@@ -311,6 +325,14 @@ void WorldSession::HandleArenaTeamRemoveOpcode(WorldPacket& recvData)
         SendArenaTeamCommandResult(ERR_ARENA_TEAM_QUIT_S, "", "", ERR_ARENA_TEAM_LEADER_LEAVE_S);
         return;
     }
+
```
Comment (Jildor) [inline src/server/game/Handlers/ArenaTeamHandler.cpp]: Whitespaces

**Comment by Jildor** [comment]
Tested and works 👍

**Comment by Gooyeth** [comment]
Thanks @Jildor 😄 

@Aokromes, has given me a better suggestion.

**Comment by tkrokli** [comment]
Very well. Please remember to link this closed PR from the new one. :-)

**Comment by Jildor** [comment]
What is it that you're plotting @Aokromes ?

**Pull request closed without merging**